### PR TITLE
Remove `_` in commands

### LINKED ISSUE #63: Missing autocomplete for commands

### LINKED ISSUE #64: Unnecessary `_` in commands affects user experience

### PR DESCRIPTION
**Pull request by marcuspang**
# Changes
1. Replace `commons-cli` with `picocli`
2. Remove `_` in commands
3. Add `@Command` annotations for all commands and remove all `XXXParser` files except `AddressBookParser`
4. Add `RootCommand` which contains all available commands in TruthTable
5. Simplify flag validation for arguments with picocli (see optional flags in `EditPersonCommand`)

# Note
1. Empty constructor is required for `@Command` files
2. Potential issue with `Index` arguments being used as they might be one-based

# Issues
1. standardise help messages, can make use of `.usage()` and define descriptions for each `@Option` or `@Parameter`
2. standardise error messages, as the format may not be consistent for invalid commands/arguments and command execution errors
6. add tests for commands

Fixes #64

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AssignTaskCommand.java [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
+    public static final String COMMAND_WORD = "assign task";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-        + ": Assigns a task to the specified member in the team.\n"
-        + "Parameters: index of task \n"
+            + ": Assigns a task to the specified member in the team.\n"
+            + "Parameters: index of task \n"
             + "Parameters: name of person \n"
-        + "Example: " + COMMAND_WORD + " 1" + " Bernice Yu";
+            + "Example: " + COMMAND_WORD + " 1" + " Bernice Yu";
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/AssignTaskCommand.java:30]: same as above, should probably update all help messages/UG at one go

File: src/main/java/seedu/address/logic/commands/AssignTaskRandomlyCommand.java [hunk 2]
```diff
@@ -6,16 +6,19 @@
...
+import picocli.CommandLine;
+import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.person.Person;
 import seedu.address.model.team.Task;
 
 /**
- * Assigns a task to a specific member in the team.
+ * Assigns a task to a random member in the team.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/AssignTaskRandomlyCommand.java:17]: lol good catch

File: src/main/java/seedu/address/logic/commands/ClearCommand.java [hunk 3]
```diff
@@ -2,22 +2,24 @@
...
 import static java.util.Objects.requireNonNull;
 
+import picocli.CommandLine;
 import seedu.address.model.AddressBook;
 import seedu.address.model.Model;
 
 /**
  * Clears the address book.
  */
+@CommandLine.Command(name = "clear")
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java:12]: KIV: do you think we should have a confirmation message?

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+
+import picocli.CommandLine;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Command that contains all subcommands starting with assign.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:10]:
```suggestion
 * Command that contains all subcommands starting with {@code assign}.
```

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 5]
```diff
@@ -1,53 +1,25 @@
...
-import seedu.address.commons.core.Messages;
-import seedu.address.commons.core.index.Index;
+import picocli.CommandLine;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
-import seedu.address.model.person.Person;
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Command that contains all subcommands starting with delete.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:10]:
```suggestion
 * Command that contains all subcommands starting with {@code delete}.
```

File: src/main/java/seedu/address/logic/commands/DeletePersonCommand.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+import picocli.CommandLine;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Deletes a person identified using it's displayed index from the address book.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/DeletePersonCommand.java:15]:
```suggestion
 * Deletes a person identified using its displayed index from the address book.
```

File: src/main/java/seedu/address/logic/commands/DeletePersonCommand.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deletePerson(personToDelete);
+        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+    }
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/DeletePersonCommand.java:46]: KIV: we should probably do a proper scan here and delete the person from all the tasks/teams that this person is associated with

File: src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java [hunk 8]
```diff
@@ -13,9 +14,10 @@
 /**
  * Deletes a task identified using it's displayed index from the task list.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java:15]:
```suggestion
 * Deletes a task identified using its displayed index from the task list.
```

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 9]
```diff
@@ -1,226 +1,22 @@
...
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
-import seedu.address.model.tag.Tag;
 
 /**
- * Edits the details of an existing person in the address book.
+ * Command that contains all subcommands starting with edit.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:10]:
```suggestion
 * Command that contains all subcommands starting with {@code edit}.
```

File: src/main/java/seedu/address/logic/commands/AddPersonCommand.java [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+@CommandLine.Command(name = "person")
+public class AddPersonCommand extends Command {
+    public static final String COMMAND_WORD = "add person";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
+            + "Parameters: "
+            + FLAG_NAME_STR + " NAME "
+            + FLAG_PHONE_STR + " PHONE "
+            + FLAG_EMAIL_STR + " EMAIL "
+            + FLAG_ADDRESS_STR + " ADDRESS "
+            + FLAG_TAG_STR + " TAG]...\n"
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/AddPersonCommand.java:41]: we're missing the opening bracket `[` here i think

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 11]
```diff
@@ -1,24 +1,23 @@
...
-import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
 
+import picocli.CommandLine;
+import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 
 /**
- * Lists all persons in the address book to the user.
+ * Command that contains all subcommands starting with list.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:10]:
```suggestion
 * Command that contains all subcommands starting with {@code list}.
```

File: src/main/java/seedu/address/logic/commands/ListMembersCommand.java [hunk 12]
```diff
@@ -26,12 +31,12 @@ public CommandResult execute(Model model) throws CommandException {
...
         return new CommandResult(
-            String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-            || (other instanceof ListMembersCommand); // instanceof handles nulls
+        return (other instanceof ListMembersCommand); // instanceof handles nulls
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/ListMembersCommand.java:39]: we can probably just do this for all the other classes too

File: src/main/java/seedu/address/logic/commands/ListTasksCommand.java [hunk 13]
```diff
@@ -1,32 +1,40 @@
...
+import picocli.CommandLine;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 
 
-
 /**
  * Lists all tasks of the current team.
  */
+@CommandLine.Command(name = "tasks")
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/ListTasksCommand.java:19]: on a sidenote, is it possible to use the `COMMAND_WORD` from below inside the annotation? if cannot then nvm i guess

File: src/main/java/seedu/address/logic/commands/MainCommand.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+/**
+ * Main command which contains all commands used in TruthTable.
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/MainCommand.java:9]: Hm maybe `RootCommand` might be a better name?
Then the docstring should also be more explanatory, e.g.

> An empty "root" command that allows pico-cli to instantiate all possible first-level commands

(idk we can probably come up with something better than what i wrote)

File: src/main/java/seedu/address/logic/commands/MainCommand.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    MarkCommand.class,
+    UnmarkCommand.class,
+    HelpCommand.class
+})
+public class MainCommand implements Runnable {
+    @CommandLine.Spec
+    private CommandLine.Model.CommandSpec spec;
+
+    public void run() {
+        throw new CommandLine.ParameterException(spec.commandLine(), MESSAGE_UNKNOWN_COMMAND);
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/MainCommand.java:31]: might be worth explaining somewhere why `run()` throws an error - e.g. 
```java
/**
 * The run() method will only be called if none of the (valid) subcommands can be instantiated.
 */
```

File: src/main/java/seedu/address/logic/parser/LocalDateTimeConverter.java [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Converter from {@code String} arguments to {@code LocalDateTime}.
+ */
+public class LocalDateTimeConverter implements CommandLine.IParameterConsumer {
+    @Override
+    public void consumeParameters(Stack<String> args, CommandLine.Model.ArgSpec argSpec,
+                                  CommandLine.Model.CommandSpec commandSpec) {
+        if (args.size() != 2) {
+            throw new CommandLine.ParameterException(commandSpec.commandLine(), "Requires date and time");
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/parser/LocalDateTimeConverter.java:19]: Maybe also include required format for date and time in the exception? e.g. `yyyy-MM-dd HH:mm (e.g. 2022-10-25 14:21)`

File: src/main/java/seedu/address/logic/parser/ParameterExceptionHandler.java [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+import picocli.CommandLine;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Used to throw the error during command execution as {@code CommandLine#execute} does not throw an Exception
+ */
+public class ParameterExceptionHandler implements CommandLine.IParameterExceptionHandler {
+    @Override
+    public int handleParseException(CommandLine.ParameterException ex, String[] args) throws Exception {
+        System.out.println("here");
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/parser/ParameterExceptionHandler.java:14]: debug print?

**Review by mjgui — DISMISSED** [review]
I'm guessing tests also haven't been fixed yet, but here's some initial comments.
Appreciate your hard work, good sir. Your nation thanks you for your service.

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 18]
```diff
@@ -1,67 +1,25 @@
...
-        return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+        throw new CommandException(MESSAGE_UNKNOWN_COMMAND);
     }
 
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof AddCommand // instanceof handles nulls
-                && toAdd.equals(((AddCommand) other).toAdd));
-    }
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:66]: Testing purposes

File: src/main/java/seedu/address/logic/commands/AddLinkCommand.java [hunk 19]
```diff
@@ -2,17 +2,23 @@
...
 import seedu.address.model.team.Link;
+import seedu.address.model.team.Url;
 
 /**
  * Adds a new link to TruthTable.
  */
+@CommandLine.Command(name = "link")
 public class AddLinkCommand extends Command {
-    public static final String COMMAND_WORD = "add_link";
+    public static final String COMMAND_WORD = "add link";
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddLinkCommand.java:21]: The thing about using picocli with the help string is that the command usage will already be formatted. Do you think we can standardise the help messages in a separate PR?

File: src/main/java/seedu/address/logic/commands/AddPersonCommand.java [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+            + "Example: " + COMMAND_WORD + " "
+            + FLAG_NAME_STR + " \"John Doe\" "
+            + FLAG_PHONE_STR + " 98765432 "
+            + FLAG_EMAIL_STR + " johnd@example.com "
+            + FLAG_ADDRESS_STR + " \"311, Clementi Ave 2, #02-25\" "
+            + FLAG_TAG_STR + " friends "
+            + FLAG_TAG_STR + " owesMoney";
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddPersonCommand.java:48]: Any point to doing this?

File: src/main/java/seedu/address/logic/parser/ParameterExceptionHandler.java (hunk 17 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/parser/ParameterExceptionHandler.java:14]:
```suggestion
```

File: src/main/java/seedu/address/logic/commands/AddPersonCommand.java (hunk 10 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddPersonCommand.java:41]:
```suggestion
            + FLAG_TAG_STR + " [TAG]...\n"
```

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java [hunk 21]
```diff
@@ -2,36 +2,31 @@
...
-    public static final String COMMAND_WORD = "add_task";
+    public static final String COMMAND_WORD = "add task";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Adds a task to the current team.\n"
-            + "Parameters: TASK_NAME\n"
-            + "Example: " + COMMAND_WORD + " merge PR#12";
+        + ": Adds a task to the current team.\n"
+        + "Parameters: name of task (cannot be empty and cannot begin with a space) \n"
+        + "Example: " + COMMAND_WORD + " merge PR#12";
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddTaskCommand.java:20]: Yes the way we parse strings now require quotes if they contain spaces

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+@CommandLine.Command(name = "assign", subcommands = {
+    AssignTaskCommand.class,
+    AssignTaskRandomlyCommand.class,
+})
+public class AssignCommand extends Command {
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        throw new CommandException(MESSAGE_UNKNOWN_COMMAND);
+    }
+
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java:21]: Doesn’t seem to be any

File: src/main/java/seedu/address/logic/commands/ListTasksCommand.java (hunk 13 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/ListTasksCommand.java:19]: Yes you can use a static variable inside the annotation even if the annotation is for the same class

File: src/main/java/seedu/address/logic/commands/AddMemberCommand.java [hunk 23]
```diff
@@ -27,21 +29,16 @@ public class AddMemberCommand extends Command {
...
+    @CommandLine.Parameters(arity = "1")
+    private Name targetPersonName;
 
-    /**
-     * Creates an AddMemberCommand to add the {@code Person} with the specified {@code Name} to the current {@code Team}
-     */
-    public AddMemberCommand(Name targetPersonName) {
-        requireNonNull(targetPersonName);
-        this.targetPersonName = targetPersonName;
+    public AddMemberCommand() {
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddMemberCommand.java:35]: I think we should standardise such commands to take in index as default parameter. If they wish to specify by name, they need to do `-n`. What do you think?

File: src/main/java/seedu/address/logic/commands/ListMembersCommand.java (hunk 12 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/ListMembersCommand.java:39]: Don’t think there are any other commands with this

File: src/main/java/seedu/address/logic/commands/AddLinkCommand.java (hunk 19 above)
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddLinkCommand.java:21]: Will be addressed in #102

File: src/main/java/seedu/address/logic/commands/AddMemberCommand.java (hunk 23 above)
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AddMemberCommand.java:35]: Will be addressed in #103

File: src/main/java/seedu/address/logic/commands/DeletePersonCommand.java (hunk 7 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/DeletePersonCommand.java:46]: This should be done at the `AddressBook` level when `deletePerson` is called.

Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/DeletePersonCommand.java:46]: Will be addressed in #104

File: src/main/java/seedu/address/logic/commands/DeletePersonCommand.java [hunk 24]
```diff
@@ -0,0 +1,55 @@
...
+ * Deletes a person identified using its displayed index from the address book.
+ */
+@CommandLine.Command(name = "person")
+public class DeletePersonCommand extends Command {
+
+    public static final String COMMAND_WORD = "delete person";
+
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Deletes the person identified by the index number used in the displayed person list.\n"
+                    + "Parameters: INDEX (must be a positive integer)\n" + "Example: " + COMMAND_WORD + " 1";
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/logic/commands/DeletePersonCommand.java:24]: We probably need a consistent way to differentiate `MEMBER_INDEX` (from the current `teamMembersList`) and the overall contacts list (`PERSON_INDEX` or `CONTACT_INDEX`)
```suggestion
                    + "Parameters: PERSON_INDEX (must be a positive integer)\n" + "Example: " + COMMAND_WORD + " 1";
```

File: src/main/java/seedu/address/logic/commands/EditPersonCommand.java [hunk 25]
```diff
@@ -0,0 +1,264 @@
...
+ */
+@CommandLine.Command(name = "person")
+public class EditPersonCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit person";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/logic/commands/EditPersonCommand.java:47]:
```suggestion
            + "Parameters: PERSON_INDEX (must be a valid positive integer) "
```

File: src/main/java/seedu/address/logic/commands/ExitCommand.java [hunk 26]
```diff
@@ -1,15 +1,17 @@
 package seedu.address.logic.commands;
 
+import picocli.CommandLine;
 import seedu.address.model.Model;
 
 /**
  * Terminates the program.
  */
+@CommandLine.Command(name = "exit")
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/logic/commands/ExitCommand.java:9]: Unrelated: can we have a 1-2 second delay before exiting the application so it feels more realistic (and users have time to read the exit message?

File: src/main/java/seedu/address/logic/commands/ExitCommand.java [hunk 27]
```diff
@@ -1,15 +1,17 @@
...
 public class ExitCommand extends Command {
 
     public static final String COMMAND_WORD = "exit";
 
-    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "Exiting Address Book as requested ...";
+    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "Exiting Address Book as requested...";
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/logic/commands/ExitCommand.java:14]:
```suggestion
    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "Exiting TruthTable as requested...";
```

File: src/main/java/seedu/address/logic/commands/MarkCommand.java [hunk 28]
```diff
@@ -4,54 +4,54 @@
...
  * Marks a specified task as complete.
  */
+@CommandLine.Command(name = "mark")
 public class MarkCommand extends Command {
     public static final String COMMAND_WORD = "mark";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
         + ": Marks the specified existing task as complete.\n"
-        + "Parameters: INDEX of task (must be a positive integer) \n"
+        + "Parameters: TASK_INDEX (must be a positive integer) \n"
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/logic/commands/MarkCommand.java:22]:
```suggestion
        + "Parameters: TASK_INDEX (must be a valid positive integer) \n"
```

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java [hunk 29]
```diff
@@ -4,48 +4,48 @@
...
  * Marks a specified task as incomplete.
  */
+@CommandLine.Command(name = "unmark")
 public class UnmarkCommand extends Command {
     public static final String COMMAND_WORD = "unmark";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
         + ": Marks the specified existing task as incomplete.\n"
-        + "Parameters: INDEX of task (must be a positive integer) \n"
+        + "Parameters: TASK_INDEX (must be a positive integer) \n"
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/logic/commands/UnmarkCommand.java:22]:
```suggestion
        + "Parameters: TASK_INDEX (must be a valid positive integer) \n"
```

**Review by Devanshshah1309 — APPROVED** [review]
Impressive work @marcuspang! Thanks for all the effort bro. 🎩 

Just some minor nitpicks

File: docs/UserGuide.md [hunk 30]
```diff
@@ -41,144 +41,148 @@ TruthTable is a **desktop app for managing software engineering teams, optimized
...
 Delete a team member from the user’s team. Will throw an error if no member with that name is found. Take note that
 names are case-sensitive.
 
 Format:
-* `delete_member TEAM_MEMBER_INDEX`
-* `delete_member --index=INDEX` (Coming soon!)
-* `delete_member -i INDEX` (Coming soon!)
+* `delete member TEAM_MEMBER_INDEX`
+* `delete member --index=INDEX` (Coming soon!)
+* `delete member -i INDEX` (Coming soon!)
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply docs/UserGuide.md:108]: Not very necessary. The other fields will be more meaningful to use

File: src/main/java/seedu/address/logic/commands/ExitCommand.java (hunk 27 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/ExitCommand.java]: Should be addressed in #79

File: src/main/java/seedu/address/logic/commands/ExitCommand.java (hunk 26 above)
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/ExitCommand.java:10]: Done inside the `MainWindow`

File: src/main/java/seedu/address/logic/commands/AssignCommand.java (hunk 22 above)
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java:21]: See #105

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 31]
```diff
@@ -22,6 +25,8 @@
 public class ParserUtil {
 
     public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
+    public static final String MESSAGE_INVALID_DATETIME = "Datetime should have the following format yyyy-MM-dd HH:mm"
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:28]:
```suggestion
    public static final String MESSAGE_INVALID_DATETIME = "DateTime should have the following format yyyy-MM-dd HH:mm"
```

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 32]
```diff
@@ -23,6 +25,8 @@
  * a menu bar and space where other JavaFX elements can be placed.
  */
 public class MainWindow extends UiPart<Stage> {
+    /** Number of seconds delay before the application closes after running this command */
+    public static final int DELAY_DURATION = 5;
```
Comment (mjgui) [inline src/main/java/seedu/address/ui/MainWindow.java:29]:
```suggestion
    public static final int DELAY_DURATION_SECONDS = 5;
```

File: src/test/java/seedu/address/logic/commands/EditTeamCommandTest.java [hunk 33]
```diff
@@ -0,0 +1,9 @@
+package seedu.address.logic.commands;
+
+import org.junit.jupiter.api.Test;
+
+class EditTeamCommandTest {
+    @Test
+    void testEquals() {
+    }
```
Comment (mjgui) [inline src/test/java/seedu/address/logic/commands/EditTeamCommandTest.java:8]: mark as todo empty? idk

File: src/main/java/seedu/address/logic/commands/SetTeamCommand.java [hunk 34]
```diff
@@ -38,7 +38,7 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Team> teamList = model.getTeamList();
         Team currentTeam = model.getTeam();
-        Team targetTeam = new Team(targetTeamName);
+        Team targetTeam = new Team(targetTeamName, "No description added");
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/logic/commands/SetTeamCommand.java:41]: Sure

**Pull request merged**